frontend/amundsen_application/static/js/components/ResourceList/ResourceListHeader/index.tsx

```diff
@@ -11,35 +11,50 @@ import {
   SOURCE_HEADER_TITLE,
   BADGES_HEADER_TITLE,
   LAST_UPDATED_HEADER_TITLE,
+  ENTITY_HEADER_TITLE,
 } from './constants';
 
 export interface ResourceListHeaderProps {
   resourceTypes: ResourceType[];
 }
 
-const contentHeaderTitle = (type: ResourceType): string => {
+const getResourceHeaders = (type: ResourceType) => {
   switch (type) {
     case ResourceType.dashboard:
-      return LAST_UPDATED_HEADER_TITLE;
-
+      return [
+        RESOURCE_HEADER_TITLE,
+        SOURCE_HEADER_TITLE,
+        LAST_UPDATED_HEADER_TITLE,
+      ];
+    case ResourceType.feature:
+      return [
+        RESOURCE_HEADER_TITLE,
+        SOURCE_HEADER_TITLE,
+        BADGES_HEADER_TITLE,
+        ENTITY_HEADER_TITLE,
+      ];
+    case ResourceType.table:
+      return [RESOURCE_HEADER_TITLE, SOURCE_HEADER_TITLE, BADGES_HEADER_TITLE];
+    case ResourceType.user:
+      return [RESOURCE_HEADER_TITLE, SOURCE_HEADER_TITLE, BADGES_HEADER_TITLE];
     default:
-      return BADGES_HEADER_TITLE;
+      return [];
   }
 };
+
 const ResourceListHeader: React.FC<ResourceListHeaderProps> = ({
   resourceTypes,
 }: ResourceListHeaderProps) => {
-  const contentHeader =
-    resourceTypes.length === 1 ? contentHeaderTitle(resourceTypes[0]) : '';
+  const headers = getResourceHeaders(resourceTypes[0]);
   return (
     <div className="resource-list-header">
-      <span className="resource">
-        <span className="resource-text">{RESOURCE_HEADER_TITLE}</span>
-      </span>
-      <span className="source">{SOURCE_HEADER_TITLE}</span>
-      <span className="badges">
-        <span className="badges-text">{contentHeader}</span>
-      </span>
+      {headers?.map((headerText, index) => {
+        return (
+          <span className={`header-${index}`} key={`header-${index}`}>
```

Comment:
Hmmm... this header-0, header-1, etc. is a bit weird and unexpected.

Would it be possible for us to create a map beforehand and give them more meaningful names?
Something like:
```
 const resourceTypeToHeaderClassMap = {
    0: 'dashboard-header',
    1: 'user-header'
    ...
```